File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res, res_a = nums[0], nums[0]
        for i, num in enumerate(nums):
            if i == 0:
                continue
            if res_a + num >= num:
                res_a = res_a + num
            else:
                res_a = num
            res = max(res, res_a)
            # print(res_a)
        return res

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> uset;
        for(int num : nums){
            uset.insert(num);
        }
        int cnt = 0, res = 0;

        for(auto elem: uset){
            cnt = 0;
            if(uset.count(elem-1) == 0){
                while(uset.count(elem++)){
                    cnt++;
                }
                res = max(res, cnt);
            }
        }
        return res;
    }
};
